### PR TITLE
ROC-8055: place ResponseMethod behind feature flag

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-8055

### Change description ###
Place ResponseMethod behind feature flag

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java [hunk 1]
```diff
@@ -37,34 +40,66 @@
...
 
-    @Before
-    public void setup() {
-        caseDetailsConverter = new CaseDetailsConverter(caseMapper,
-            JsonMapperFactory.create(),
-            new WorkingDayIndicator(publicHolidaysCollection, nonWorkingDaysCollection),
-            33);
-    }
+    @Nested
+    class CtscSDisabledTests {
```
Comment (akritis22) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java:44]: remove extra "S" in name

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java [hunk 2]
```diff
@@ -37,34 +40,66 @@
...
-        assertThat(claim.getState().getValue()).isEqualTo(caseDetails.getState());
+        @Test
+        void convertsCaseDetailsToClaimWithIntentionToProceedDeadline() {
+            CaseDetails caseDetails = successfulCoreCaseDataStoreSubmitResponse();
+            Claim claim = caseDetailsConverter.extractClaim(caseDetails);
+            assertThat(claim.getIntentionToProceedDeadline()).isNotNull();
+        }
+
+        @Test
+        void convertsCaseDetailsToClaimWithResponseMethod() {
```
Comment (akritis22) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java:79]: Please change name to include "without" instead or "with

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java (hunk 1 above)
Comment (daveoh) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java]: Done

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java (hunk 2 above)
Comment (daveoh) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java]: Done

File: src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java [hunk 3]
```diff
@@ -37,34 +40,66 @@
...
+                new WorkingDayIndicator(publicHolidaysCollection, nonWorkingDaysCollection),
+                33,
+                true);
+        }
+
+        @Test
+        void convertsCaseDetailsToClaimWithResponseMethod() {
+            CaseDetails caseDetails = successfulCoreCaseDataStoreSubmitResponseWithDQ();
+            Claim claim = caseDetailsConverter.extractClaim(caseDetails);
+            assertThat(claim.getResponse().get().getResponseMethod().get()).isEqualTo(ResponseMethod.DIGITAL);
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverterTest.java:102]: assert4j has a `.contains(expectedValue)` for optionals

**Pull request merged**